libs/forecasting/src/lib/ForecastingDashboard/Controls/ForecastComparison.tsx

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Stack, Text } from "@fluentui/react";
+import {
+  ModelAssessmentContext,
+  defaultModelAssessmentContext,
+  BasicHighChart,
+  JointDataset,
+  orderByTime
+} from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
+import { SeriesOptionsType } from "highcharts";
+import React from "react";
+
+import { forecastingDashboardStyles } from "../ForecastingDashboard.styles";
+
+import { getForecastPrediction } from "./getForecastPrediction";
+
+export class IForecastComparisonProps {}
+
+export interface IForecastComparisonState {
+  timeSeriesId?: number;
+  baselinePrediction?: number[];
+}
+
+const stackTokens = {
+  childrenGap: "l1"
+};
+
+export class ForecastComparison extends React.Component<
+  IForecastComparisonProps,
+  IForecastComparisonState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public constructor(props: IForecastComparisonProps) {
+    super(props);
+    this.state = {
+      baselinePrediction: undefined
+    };
+  }
+
+  public async componentDidMount(): Promise<void> {
+    const baselinePrediction = await this.getBaselineForecastPrediction();
+    if (baselinePrediction) {
+      this.setState({ baselinePrediction });
+    }
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    // Check if the time series was changed.
+    // In that case, we need to update our state accordingly.
+    const currentlySelectedTimeSeriesId =
+      this.context.baseErrorCohort.cohort.getCohortID();
+    if (currentlySelectedTimeSeriesId !== this.state.timeSeriesId) {
+      const baselinePrediction = await this.getBaselineForecastPrediction();
+      this.setState({
+        baselinePrediction,
+        timeSeriesId: currentlySelectedTimeSeriesId
+      });
+    }
+  }
+
+  public render(): React.ReactNode {
+    const classNames = forecastingDashboardStyles();
+
+    const indices = this.context.dataset.index;
+    if (this.context === undefined || indices === undefined) {
+      return;
+    }
+    const rowIndices = this.context.baseErrorCohort.cohort.filteredData.map(
+      (datum) => indices[datum.Index]
+    );
+
+    let trueY: SeriesOptionsType[] = [];
+    if (this.context.jointDataset.numLabels === 1) {
+      trueY = [
+        {
+          data: orderByTime(
```

Comment:
sorting is usually intense, is it worthy to use state to cache?

Reply:
Done!